### ISSUE TITLE
resource not in expected location

### ISSUE THREAD

**Issue by dpgrote** [spack/spack#11666]
When running "spack install warpx", the resources that are downloaded are not put in the expected destinations. This is only a problem on the develop branch. In the master branch, this works correctly, and worked correctly in develop in the past.

The warpx/package.py file has the following specification:

```
    resource(name='amrex',
             git='https://github.com/AMReX-Codes/amrex.git',
             tag='master',
             destination='.')

    resource(name='picsar',
             git='https://bitbucket.org/berkeleylab/picsar.git',
             tag='master',
             destination='.')
```

Here are the relevant lines in the output:

```
==> Already staged resource-amrex-m2gvfqh7nrr4zo4yvuf6t6kzvovcg6ze in /Users/grote1/packages/spack-dpgrote/var/spack/stage/resource-amrex-m2gvfqh7nrr4zo4yvuf6t6kzvovcg6ze
==> Moving resource stage
	source : /Users/grote1/packages/spack-dpgrote/var/spack/stage/resource-amrex-m2gvfqh7nrr4zo4yvuf6t6kzvovcg6ze/src/
	destination : /Users/grote1/packages/spack-dpgrote/var/spack/stage/warpx-master-m2gvfqh7nrr4zo4yvuf6t6kzvovcg6ze/src/./src
==> Already staged resource-picsar-m2gvfqh7nrr4zo4yvuf6t6kzvovcg6ze in /Users/grote1/packages/spack-dpgrote/var/spack/stage/resource-picsar-m2gvfqh7nrr4zo4yvuf6t6kzvovcg6ze
```

The amrex source is put into src/src instead of amrex, and picsar is not copied at all.

### Information on your system

This happens on both mac and linux.

**Comment by mwkrentel** [comment]
See #11647.  I'm guessing that many, if not all packages that use a
resource are currently broken.

So, I assume this is fairly high priority.

**Comment by mwkrentel** [comment]
@tgamblin  @tldahlgren 
I looked into this a bit more.  I found that my package (intel-xed)
uses `destination` and breaks because spack fails to move the resource
directory as it used to.

```
resource(name='mbuild',
         git='https://github.com/intelxed/mbuild.git',
         branch='master', destination='',
         when='@develop')
```

Other packages use `placement` instead of `destination` and some of
them seem to work.  Some use both.

Should I change my package to use placement instead of destination?

Or, should I wait for something to get fixed where the current recipe
will work again?

What is the preferred usage?

Read the docs only mentions destination, not placement.
And `lib/spack/spack/resource.py` doesn't have anything useful.
